### PR TITLE
Feature/form validation

### PR DESCRIPTION
**Pull request by luiszaval**
- import new packages Formik and Yup in order for all app forms to now handle input validation
- Formik handles all of our previously explicit state that had been moved from redux
- Yup deals with error handling
- Moved all individual modal css files to one file under custom modal
- Deleted custom form component (formik now handles)
- Combined register and login css under one file (might be a good idea to create a pages folder under which these files can live)
- Updated login and register css to match timmy's previous work on login
- Updated modals presentation to use similar aspects of slack

### CODE REVIEW

File: src/components/Modals/InviteMembersModal/InviteMembersModal.js [hunk 1]
```diff
@@ -15,76 +20,85 @@ const mapStateToProps = (state)=>{
...
+                        </Form>
+                    )}
+                </Formik>
+            </>
         );
         return (
             <CustomModal 
                 show={showInviteMembersModal} 
-                onHide={this.handleHide} 
+                onHide={()=>handleInviteMembersModal(false)}
```
Comment (BitPhoenix) [inline src/components/Modals/InviteMembersModal/InviteMembersModal.js:101]: Spacing () => handle...

File: src/components/Modals/OrgSettingsModal/OrgSettingsModal.js [hunk 2]
```diff
@@ -41,20 +43,22 @@ class OrgSettingsModal extends Component {
...
-                        type="button" 
-                        onClick={()=>this.handleInviteMembers(true)}
-                        >Invite new members to your Org
-                    </CustomButton>
-                    <InviteMembersModal />
-                </CustomForm> 
+                        />
+                        <CustomButton 
+                            type="button" 
+                            onClick={()=>this.handleInviteMembers(true)}
```
Comment (BitPhoenix) [inline src/components/Modals/OrgSettingsModal/OrgSettingsModal.js:56]: Spacing () => this

File: src/components/Modals/PendingInvitationsModal/PendingInvitationsModal.js [hunk 3]
```diff
@@ -59,14 +55,14 @@ class PendingInvitationsModal extends Component {
...
+                <form className={customForm}>
                     {invitationsDisplay}
-                </CustomForm>
+                </form>
             );
         return (
             <CustomModal
                 show={showPendingInvitationsModal && invitations.length > 0}
-                onHide={this.handleHide} 
+                onHide={()=>handlePendingInvitationsModal(false)}
```
Comment (BitPhoenix) [inline src/components/Modals/PendingInvitationsModal/PendingInvitationsModal.js:65]: Spacing () =>

File: src/components/Register/Register.js [hunk 4]
```diff
@@ -1,111 +1,125 @@
...
+        const { changeRoute } = this.props
+        const form = (
+            <>
+                <Formik
+                    initialValues={{
+                    username: '',
+                    password: '',
+                    confirmPassword: '',
+                    }}
+                    validationSchema={Yup.object({
```
Comment (BitPhoenix) [inline src/components/Register/Register.js:40]: Extract schema

File: src/components/Register/Register.js [hunk 5]
```diff
@@ -1,111 +1,125 @@
...
+                            .email('Invalid Email Address')
+                            .required('Required'),
+                    password: Yup.string()
+                        .required('No password provided')
+                        .min(8, 'Password is too short - should be 8 characters minimum.')
+                        .matches(/(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]/, 'Password must be at least one letter, one number and one special character'),
+                    confirmPassword: Yup.string()
+                        .oneOf([Yup.ref('password'), null], 'Passwords must match')
+                    })}
+                    onSubmit={(values, {setSubmitting, setStatus}) =>{
```
Comment (BitPhoenix) [inline src/components/Register/Register.js:51]: Extra onSubmit

File: src/components/Register/Register.js [hunk 6]
```diff
@@ -1,111 +1,125 @@
...
+                                placeholder="confirm password"
+                            />
+                            <div className={btns}>
+                                <CustomButton 
+                                type='submit'
+                                >Submit
+                                </CustomButton>
+                                <CustomButton
+                                type="button"
+                                onClick = {()=> changeRoute({path:"/login"})}
```
Comment (BitPhoenix) [inline src/components/Register/Register.js:96]: () =>

File: src/components/UI/CustomFormInput/FormInput.module.css [hunk 7]
```diff
@@ -20,12 +20,24 @@
...
 }
 
+.nameDisplay{
+    width: 70%;
+    margin: 0;
+}
+
+.error{
+    color: red;
+}
```
Comment (BitPhoenix) [inline src/components/UI/CustomFormInput/FormInput.module.css:37]: Space after class name

File: src/components/UI/CustomModal/CustomModal.module.css [hunk 8]
```diff
@@ -23,5 +23,98 @@
...
+}
+
+.usernameDisplay {
+    background: white;
+    color: black;
+    padding: 5px;
+    border-radius: 5px;
+    margin-right: 5px;
+    width: auto;
+}
```
Comment (BitPhoenix) [inline src/components/UI/CustomModal/CustomModal.module.css:56]: Any classes specific to a certain modal should probably go outside of the main CustomModal.module.css

**Review by BitPhoenix — CHANGES_REQUESTED** [review]
Left first pass comments!

File: src/components/Login/Login.js [hunk 9]
```diff
@@ -34,88 +34,82 @@ class Login extends Component {
...
+                    initialValues={{
+                    username: '',
+                    password: '',
+                    }}
+                    validationSchema={Yup.object({
+                    username: Yup.string()
+                        .required('Required'),
+                    password: Yup.string()
+                        .required('No password provided') ,
+                    })}
```
[the remark this answers is not in the document]
Comment (luiszaval) [reply src/components/Login/Login.js:53]: Yeah I know right. Super intuitive and readable

File: src/components/Login/Login.js [hunk 10]
```diff
@@ -34,88 +34,82 @@ class Login extends Component {
...
+        const { register, create, main, registerForm, content, btns, bottomBorder, logoCol, greenColor, logo, formCol } = styles;
+        const {errorMsg, customForm } = formStyles 
+        const form = (
+            <>
+                <Formik
+                    initialValues={{
+                    username: '',
+                    password: '',
+                    }}
+                    validationSchema={Yup.object({
```
[the remark this answers is not in the document]
Comment (luiszaval) [reply src/components/Login/Login.js:48]: Good idea. WIll do

File: src/components/Login/Login.js [hunk 11]
```diff
@@ -1,31 +1,31 @@
...
+import { Formik, Form } from "formik";
 
-import styles from "./Login.module.css";
-import classes from "../Register/Register.module.css"
-import { Container, Row, Col } from 'react-bootstrap'
+import CustomButton from '../UI/CustomButton/CustomButton';
+import CustomFormInput from '../UI/CustomFormInput/FormInput';
+
+import formStyles from '../UI/CustomModal/CustomModal.module.css'
+import styles from "../Register/Register.module.css"
```
[the remark this answers is not in the document]
Comment (luiszaval) [reply src/components/Login/Login.js:11]: Exactly what I was thinking up in my PR comments

File: src/components/Login/Login.js [hunk 12]
```diff
@@ -34,88 +34,82 @@ class Login extends Component {
...
-                            </Col>
-                        </Row>
-                    </Container>
-                </div>
-        	)
+        const { changeRoute } = this.props;
+        const { register, create, main, registerForm, content, btns, bottomBorder, logoCol, greenColor, logo, formCol } = styles;
+        const {errorMsg, customForm } = formStyles 
+        const form = (
+            <>
```
[the remark this answers is not in the document]
Comment (luiszaval) [reply src/components/Login/Login.js:42]: its shorthand for fragment:

https://reactjs.org/docs/fragments.html

fragments are recommended over divs:

https://stackoverflow.com/questions/47761894/why-are-fragments-in-react-16-better-than-container-divs

[the remark this answers is not in the document]
Comment (BitPhoenix) [reply src/components/Login/Login.js:42]: Nice! I knew about fragments but wasn't familiar with the short hand syntax. Awesome to know!

File: src/components/Modals/CreateChannelModal/CreateChannelModal.js [hunk 13]
```diff
@@ -21,140 +27,131 @@ const mapActionsToProps = {
...
+                                name="channelName"
+                                type="text"
+                                placeholder="#programming"
+                            />
+                            <FieldArray
+                                name="privateUsers"
+                                >
+                                {({insert, remove, push}) =>(
+                                    <div className={inviteMembersDisplay}>
+                                        {values.isPrivate && values.privateUsers && values.privateUsers.length > 0 ? (
```
[the remark this answers is not in the document]
Comment (luiszaval) [reply src/components/Modals/CreateChannelModal/CreateChannelModal.js:97]: yep got it makes a lot more sense

**Pull request merged**